### PR TITLE
Spike support for block device option

### PR DESCRIPTION
**Pull request by junsunchoi**
Support for running spike with block device option when typing "marshal launch"

### CODE REVIEW

File: wlutil/launch.py [hunk 1]
```diff
@@ -32,22 +32,27 @@ def get_free_tcp_port():
...
         spikeBin = str(config['spike'])
     else:
         spikeBin = 'spike'
 
     cmd = [spikeBin,
-           config.get('spike-args', ''),
+           spikeArgs,  # config.get('spike-args', ''),
            ' -p' + str(config['cpus']),
            ' -m' + str(int(config['mem'] / (1024*1024)))]
-
```
Comment (abejgonzalez) [inline wlutil/launch.py:50]: Nit: Keep the space.

File: wlutil/launch.py
[the remark this answers is not in the document]
Comment (junsunchoi) [reply wlutil/launch.py:1]: Can you check the changes?

File: docs/source/internal/Build.rst [hunk 2]
```diff
@@ -4,10 +4,12 @@ Build Process
 
 The goal of building a workload is to produce a working boot binary and
 (optionally) a root filesystem to boot from. The same outputs are used for
-Spike, Qemu, and FireSim. The one exception is that Spike does not support a
-disk, so users may choose to create an initramfs-only version of their workload
-for Spike (that binary will boot on Qemu and FireSim as well). The build process
-proceeds as follows:
+Spike, Qemu, and FireSim. Spike now supports a disk. In order to use a disk,
+libspikedevices.so should be installed by chipyard to the RISCV library path.
```
Comment (abejgonzalez) [inline docs/source/internal/Build.rst:8]:
```suggestion
``libspikedevices.so`` should be installed by Chipyard to the ``RISCV`` library path (i.e. ``$RISCV/lib`).
```

File: docs/source/internal/Build.rst [hunk 3]
```diff
@@ -4,10 +4,12 @@ Build Process
...
+Iceblk will be used as the default device. Users may also choose to create
```
Comment (abejgonzalez) [inline docs/source/internal/Build.rst:9]:
```suggestion
The Iceblk block device model will be used as the default block device model. Users may also choose to create 
```

**Review by abejgonzalez — APPROVED** [review]
Small nits. Looks good though.

File: docs/source/internal/Build.rst [hunk 4]
```diff
@@ -4,10 +4,12 @@ Build Process
 
 The goal of building a workload is to produce a working boot binary and
 (optionally) a root filesystem to boot from. The same outputs are used for
-Spike, Qemu, and FireSim. The one exception is that Spike does not support a
-disk, so users may choose to create an initramfs-only version of their workload
-for Spike (that binary will boot on Qemu and FireSim as well). The build process
-proceeds as follows:
+Spike, Qemu, and FireSim. Spike now supports a disk. In order to use a disk,
+``libspikedevices.so`` should be installed by Chipyard to the ``RISCV`` library path (i.e. ``$RISCV/lib`).
```
Comment (abejgonzalez) [inline docs/source/internal/Build.rst:8]:
```suggestion
``libspikedevices.so`` should be installed by Chipyard to the ``RISCV`` library path (i.e. ``$RISCV/lib``).
```

**Review by abejgonzalez — APPROVED** [review]

**Pull request merged**